### PR TITLE
Add error logging to flutter generate

### LINKED ISSUE #29812: Code generation should print severe logs from build_runner
Currently all logs are hidden unless we're running in verbose mode

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Ensure that the error_cache generated by build runner is surfaced in the command line. It seems like the logging interface does not forward errors, so we need to look in the file system.

Fixes https://github.com/flutter/flutter/issues/29812

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/generate.dart [hunk 1]
```diff
@@ -34,12 +33,34 @@ class GenerateCommand extends FlutterCommand {
...
         break;
       }
     }
+    // Check for errors.
+    final Directory buildDirectory = flutterProject.dartTool.childDirectory('build');
+    final Directory errorCacheParent = buildDirectory.listSync().firstWhere((FileSystemEntity entity) {
+      return entity is Directory && entity.childDirectory('error_cache').existsSync();
+    }, orElse: () => null);
+    if (errorCacheParent == null) {
+      return null;
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/commands/generate.dart]: Should this return `FlutterCommandResult` with `ExitStatus.success`?

File: packages/flutter_tools/lib/src/commands/generate.dart [hunk 2]
```diff
@@ -34,12 +33,34 @@ class GenerateCommand extends FlutterCommand {
...
+        final List<Object> stackData = errorData[1];
+        printError(errorData.first);
+        printError(stackData[0]);
+        printError(stackData[1]);
+        printError(StackTrace.fromString(stackData[2]).toString());
+      } catch (err) {
+        printError('Error reading error in ${errorFile.path}');
+      }
+    }
     return null;
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/commands/generate.dart]: `ExitStatus.fail`?

**Review by yjbanov — APPROVED** [review]

File: packages/flutter_tools/lib/src/commands/generate.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/commands/generate.dart]: null is the same as success

File: packages/flutter_tools/lib/src/commands/generate.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/commands/generate.dart]: Done

**Pull request merged**